mozilla/learning.mozilla.org
# Clubform

Implements https://github.com/mozilla/learning.mozilla.org/issues/1857, fixes https://github.com/mozilla/learning.mozilla.org/issues/1884

learning site:
- [x] make "mozilla clubs" use the same purple nav colour
- [x] only allowed logged in users to fill in the application form.
- [x] implement the UI for "when users are not logged in"
- [x] implement club form page 1
- [x] implement club form page 2
- [x] implement club form acknowledgement
- [x] implement club form send-to-teach-API functionality in line with current data
  - [x] implement club form send-to-teach-API functionality extended for the new form (*teach-api)
- [x] have the application form highlight as if we're on the mozilla clubs page in sidebar
- [x] rewrite "apply" buttons to point to new club form

teach-api:
- [x] update the club model to cover all the new fields from the application form

And once the PR is good to go:
- [x] remove login hardwire from the `teach-api.js` file (controlled by `emulateLogin`)

## follow up
- remove the old "apply" modal code from base clubs form  https://github.com/mozilla/learning.mozilla.org/issues/2045
- update the tests to cover the new application form https://github.com/mozilla/learning.mozilla.org/issues/2046
- determine whether we want localisation on the application form (since we have no staff able to review arbitrary language submissions right now) https://github.com/mozilla/learning.mozilla.org/issues/2049

## test setup
- set up a local [id.wmo](https://github.com/mozilla/id.webmaker.org), [login.wmo](https://github.com/mozilla/login.webmaker.org), [teach-api](https://github.com/mozilla/teach-api) and [learn.moz](https://github.com/mozilla/learning.mozilla.org), with the latter set to this branch (`clubform`)
  - this might require setting up a working postgres instance for id.wmo, too... please see https://github.com/mozilla/learning.mozilla.org/wiki#bootstrapping-idwebmakerorg for instructions
  - make sure to run the teach-api's `manage.py migrate` at least once, so that you're on the new form schema
  - create an `.env` file in the learn.moz root directory (if you don't already have one) and add `export TEACH_API_URL=http://localhost:8000` to that, so that the learning site uses your local copy of the teach-api
  - in teach-api, edit `./teach/settings.py`:
```
IDAPI_URL = os.environ.get('IDAPI_URL', 'http://localhost:1234')
IDAPI_CLIENT_ID = os.environ.get('IDAPI_CLIENT_ID', 'test')
IDAPI_CLIENT_SECRET = os.environ.get('IDAPI_CLIENT_SECRET', 'test')

LOGINAPI_URL = os.environ.get('LOGINAPI_URL', 'http://localhost:3000')
LOGINAPI_AUTH = os.environ.get('LOGINAPI_AUTH')

TEACH_SITE_URL = os.environ.get('TEACH_SITE_URL', 'http://localhost:8008')
```

## test steps
1. start all services
2. head over to http://localhost:8008/clubs
3. sign in (or if this is a clean install of id/login, sign up)
4. fill in the new application club form by going to [http://localhost:8008/en-US/clubs](http://localhost:8008/en-US/clubs) and clicking the "apply to be a club captain" button
  - there are two pages of fields; not filling in certain fields should prevent you from moving on to the next page/submitting, flagging fields in red with messages on what still needs to be done at the bottom of the page (try submitting page 1 and page 2 before filling anything in to confirm this)
5. submit club form with your console/network tab open in dev tools (in case things don't work!)
6. submitting should work and take you to a third screen informing you that you you submitted your application.
7. point a new tab to the teach-api admin view on [http://localhost:8000/admin](http://localhost:8000/admin) (if you get a login, please see https://github.com/mozilla/teach-api#making-a-staff-account-to-use-the-admin-route)
8. verify the club application exists
9. click the club name to see a detailed information view; verify it looks like what you filled in (it should)
10. point a new tab to [http://localhost:8008/clubs](http://localhost:8008/clubs), and scroll down to the map
11. verify your club shows up on the map
